integrations/cdi/jpa-cdi/src/main/java/io/helidon/integrations/cdi/jpa/PersistenceExtension.java

```diff
@@ -1379,12 +1382,19 @@ private static JtaExtendedEntityManager produceJtaExtendedEntityManager(Instance
             }
         }
         SynchronizationType finalSyncType = syncType == null ? SYNCHRONIZED : syncType;
+        Config config = getOrDefault(instance.select(Config.class), selectionQualifiers);
+        Map<String, Object> stackedProperties =
+            new StackedMap<>(List.of(properties.keySet()::stream,
+                                     () -> stream(config.getPropertyNames()::spliterator,
```

Comment:
Not sure there's a good alternative, given that annoyingly (but understandably)  `Config#getPropertyNames` is [not guaranteed](https://github.com/eclipse/microprofile-config/blob/8e5f457bf1a16ff631f4ced0486e7a50e53247a9/api/src/main/java/org/eclipse/microprofile/config/Config.java#L235-L239) to be complete. Practically speaking the config sources present might indeed behave well.

If there were a good way to anticipate what the interesting property names would be--if present--so the code could ask for values explicitly that'd work around this corner case. I do not pretend to know all the details of what the code is accomplishing here to know if that's feasible.

Reply:
Right. The particular application in this case is: we are making a `Map` for supplying to a specific clunky JPA API (the persistence properties supplied to `createEntityManager`). The fact that these properties are represented in the JPA API as a `Map` is a pretty bad design choice, since for virtually all use cases other than logging or debugging the `Map` will be used by the persistence provider only as a device to "probe" for a value for a particular property known to it. Because of the contract of `Map`, however, we need to report its `keySet` and its `size` and so on, even though for most use cases this will never be called/used (the persistence provider will be looking for specific properties, so will, realistically, be calling `get(String)` and not much else). So not much else to do here, I think.